app/screens/invite/invite_actions.ts

```diff
@@ -0,0 +1,395 @@
+                userId,
+                reason: formatMessage({
+                    id: 'invite.guests.error_adding_to_team',
+                    defaultMessage: 'Unable to add the guest to the team',
+                }),
+            },
+        };
+    }
+
+    const modelPromises: Array<ReturnType<typeof addMembersToChannel>> = [];
+
+    for (const {id: channelId} of channels) {
+        const addMemberToChannel = addMembersToChannel(serverUrl, channelId, [userId]);
+        modelPromises.push(addMemberToChannel);
+    }
+
+    const memberOfChannels = await Promise.all(modelPromises);
+
+    for (const {error} of memberOfChannels) {
+        if (error) {
+            return {
+                notSent: {
+                    userId,
+                    reason: formatMessage({
+                        id: 'invite.guests.error_adding_to_channels',
+                        defaultMessage: 'Unable to add the guest to the channels',
+                    }),
+                },
+            };
+        }
+    }
+
+    if (memberOfAny) {
+        return {
+            sent: {
+                userId,
+                reason: formatMessage(
+                    {
+                        id: 'invite.guests.some_channels_member',
+                        defaultMessage: 'This person is already a team member. Invited to {count} {count, plural, one {channel} other {channels}}',
+                    },
+                    {count: channels.length - memberOfAny},
+                ),
+            },
+        };
+    }
+
+    return {
+        sent: {
+            userId,
+            reason: formatMessage(
+                {
+                    id: 'invite.guests.added_to_channels',
+                    defaultMessage: 'Invited as a member of {teamDisplayName} and {count} {count, plural, one {channel} other {channels}}',
+                },
+                {
+                    teamDisplayName,
+                    count: channels.length,
+                },
+            ),
+        },
+    };
+};
+
+export const sendGuestsInvites = async (
+    serverUrl: string,
+    teamId: string,
+    teamDisplayName: string,
+    invites: Invites,
+    channels: Channel[],
+    message: string,
+    intl: IntlShape,
+    isAdmin?: boolean,
+) => {
+    const {formatMessage} = intl;
+    const {userIds, emails} = getInvitesByTypes(invites);
+
+    const sent: InviteResult[] = [];
+    const notSent: InviteResult[] = [];
+
+    const {database} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+    const modelPromises: Array<Promise<void>> = [];
+    const channelsMembers: {[channelId: string]: UserModel[]} = {};
+
+    for (const {id: channelId} of channels) {
+        const usersOnChannel = queryUsersOnChannel(database, channelId).fetch().then(
+            (users) => {
+                channelsMembers[channelId] = users;
+            });
+        modelPromises.push(usersOnChannel);
+    }
+
+    await Promise.all(modelPromises);
```

Comment:
Wonder if this is prone to race conditions. In this context, the `then` would be executed before the promise is fulfilled, or after the promise is fulfilled?

If it is after, some `then` may potentially be enqueued after the next statement. But maybe I am overthinking.